comp/trace/config/config_test.go

```diff
@@ -2517,6 +2517,8 @@ func TestMockConfig(t *testing.T) {
 		fx.Supply(corecomp.Params{}),
 		corecomp.MockModule(),
 		MockModule(),
+		// disable fetching the hostname from the core agent
+		fx.Replace(corecomp.MockParams{Overrides: map[string]interface{}{"serverless.enabled": true}}),
```

Comment:
This is a bit hacky as it relies on a side effect that might change in the future. Plus, the serverless mode is an exception and doesn't reflect how the trace-agent works in most scenarios.
I see the test does the same for `DD_SITE`. Can we just not use the hostname as example here?

Reply:
Honestly I'm only editing this test so that it stops failing when running locally, and this was the simplest fix I could find which wouldn't change what is actually tested
I do agree that it is hacky, if just removing the `hostname` test works for you then fine by me 😄 
I'll update the PR

Reply:
Thank you for fixing it in the first place and for updating the PR!